diff --git a/src/FEP/compute_fep.cpp b/src/FEP/compute_fep.cpp
index c444e8352a..116fc69ade 100644
--- a/src/FEP/compute_fep.cpp
+++ b/src/FEP/compute_fep.cpp
@@ -218,8 +218,7 @@ void ComputeFEP::init()
 
       // if pair hybrid, test that ilo,ihi,jlo,jhi are valid for sub-style
 
-      if ((strcmp(force->pair_style, "hybrid") == 0 ||
-           strcmp(force->pair_style, "hybrid/overlay") == 0)) {
+      if (utils::strmatch(force->pair_style, "^hybrid")) {
         auto *pair = dynamic_cast<PairHybrid *>(force->pair);
         for (i = pert->ilo; i <= pert->ihi; i++)
           for (j = MAX(pert->jlo, i); j <= pert->jhi; j++)
diff --git a/src/FEP/fix_adapt_fep.cpp b/src/FEP/fix_adapt_fep.cpp
index 1f311dae70..514cbbcae3 100644
--- a/src/FEP/fix_adapt_fep.cpp
+++ b/src/FEP/fix_adapt_fep.cpp
@@ -304,8 +304,7 @@ void FixAdaptFEP::init()
 
       // if pair hybrid, test that ilo,ihi,jlo,jhi are valid for sub-style
 
-      if (ad->pdim == 2 && (strcmp(force->pair_style,"hybrid") == 0 ||
-                            strcmp(force->pair_style,"hybrid/overlay") == 0)) {
+      if (ad->pdim == 2 && utils::strmatch(force->pair_style, "^hybrid")) {
         auto *pair = dynamic_cast<PairHybrid *>(force->pair);
         for (i = ad->ilo; i <= ad->ihi; i++)
           for (j = MAX(ad->jlo,i); j <= ad->jhi; j++)
diff --git a/src/INTERLAYER/pair_kolmogorov_crespi_z.cpp b/src/INTERLAYER/pair_kolmogorov_crespi_z.cpp
index 08bf44b37d..fdd85de4fd 100644
--- a/src/INTERLAYER/pair_kolmogorov_crespi_z.cpp
+++ b/src/INTERLAYER/pair_kolmogorov_crespi_z.cpp
@@ -203,7 +203,7 @@ void PairKolmogorovCrespiZ::allocate()
 void PairKolmogorovCrespiZ::settings(int narg, char **arg)
 {
   if (narg != 1) error->all(FLERR, "Illegal pair_style command");
-  if (strcmp(force->pair_style, "hybrid/overlay") != 0)
+  if (!utils::strmatch(force->pair_style, "^hybrid/overlay"))
     error->all(FLERR, "ERROR: requires hybrid/overlay pair_style");
 
   cut_global = utils::numeric(FLERR, arg[0], false, lmp);
